src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -656,8 +657,14 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 // validate name if we have one
                 if (name != null)
                 {
-                    // PROTOTYPE: check for a case "ItemX" where X is not i
+                    elementNameCount++;
 
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
```

Comment:
Support for tuples >7 will come later. We definitely need to "handle" case of 7+ without a crash though. 

We should handle 7+ tuples the same as we handle < 2 tuples at the beginning of this method and have a test. 
(I should've thought about this when I adding handling for < 2)